src/main/java/com/notably/logic/suggestion/commands/OpenSuggestionCommand.java

```diff
@@ -0,0 +1,117 @@
+package com.notably.logic.suggestion.commands;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.logic.suggestion.SuggestionCommand;
+import com.notably.model.Model;
+import com.notably.model.block.BlockTreeItem;
+import com.notably.model.suggestion.SuggestionItem;
+import com.notably.model.suggestion.SuggestionItemImpl;
+
+/**
+ * Represents a suggestion command object to open a note.
+ */
+public class OpenSuggestionCommand implements SuggestionCommand {
+    public static final String COMMAND_WORD = "open";
+    public static final String RESPONSE_MESSAGE = "Open a note";
+
+    private AbsolutePath path;
+
+    public OpenSuggestionCommand(AbsolutePath path) {
+        Objects.requireNonNull(path);
+        this.path = path;
+    }
+
+    @Override
+    public void execute(Model model) {
+        // Nullity check
+        Objects.requireNonNull(model);
+
+        // Set response text
+        model.setResponseText(RESPONSE_MESSAGE);
+
+        // Set suggestions
+        List<AbsolutePath> possiblePaths = getPossiblePaths(model);
+        List<SuggestionItem> suggestions = getSuggestions(possiblePaths, model);
+
+        setSuggestionsToModel(model, suggestions);
+    }
+
+
+    public void setSuggestionsToModel(Model model, List<SuggestionItem> suggestions) {
+        Objects.requireNonNull(model);
+        Objects.requireNonNull(suggestions);
+
+        model.setSuggestions(suggestions);
+    }
```

Comment:
Actually, my earlier point applies here too 😅
I think it's clearer if you just pass `model.setSuggestions(suggestions)` directly in `execute`? It's a one-liner in the end of the day